multiapps-controller-persistence/src/main/java/org/cloudfoundry/multiapps/controller/persistence/query/providers/SqlFileQueryProvider.java

```diff
@@ -226,14 +226,18 @@ protected SqlFileQueryProvider(String tableName, DataSourceDialect dataSourceDia
 
     }
 
-    public SqlQuery<Integer> getDeleteBySpaceQuery(String space) {
+    public SqlQuery<Integer> getDeleteBySpacesQuery(List<String> spaces) {
         return (Connection connection) -> {
             PreparedStatement statement = null;
             try {
                 statement = connection.prepareStatement(getQuery(DELETE_FILES_BY_SPACE));
-                statement.setString(1, space);
-                int deletedFiles = statement.executeUpdate();
-                logger.debug(MessageFormat.format(Messages.DELETED_0_FILES_WITH_SPACE_1, deletedFiles, space));
+                addSpacesAsBatches(statement, spaces);
+                int[] batchResults = statement.executeBatch();
+                int deletedFiles = 0;
+                for (int batchResult : batchResults) {
+                    deletedFiles += batchResult;
+                }
```

Comment:
Could be replaced by this one-liner: `int deletedFiles = IntStream.of(batchResult).sum();`